snapcraft/internal/errors.py

```diff
@@ -25,6 +25,16 @@
     'minLength': 'minimum length is {validator_value}',
 }
 
+# dict with nice error messages for invalid names in parts, hooks and apps
+_PROPERTY_ERROR_MESSAGES = {
+    "parts": "Part name does not match "
+    "'^(?!plugins$)[a-z0-9][a-z0-9+-\\/]*$'",
+    "apps": "App name does not match "
+    "'^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$'",
+    "hooks": "Hook name does not match "
+    "'^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$'",
```

Comment:
I'm not the biggest fan of this. The regex is defined in the schema, we duplicate the regex here, they easily get out of sync. Yes, we have this exact same problem if we start using regular english to represent the error. I suggest we figure out a way to maintain validation error messages in the schema itself, something [like this](https://github.com/snapcore/snapcraft/blob/master/schema/snapcraft.yaml#L215). Thoughts?